celesta-core/src/main/java/ru/curs/celesta/CallContext.java

```diff
@@ -104,35 +117,47 @@ public String getUserId() {
     }
 
     /**
-     * Commits the current transaction.
+     * Commits the current transaction. Will cause error for not-activated or closed context.
      * <p>
      * Wraps SQLException into CelestaException.
      */
     public void commit() {
-        try {
-            conn.commit();
-        } catch (SQLException e) {
-            throw new CelestaException("Commit unsuccessful: %s", e.getMessage());
+        if (state == State.ACTIVE) {
+            try {
+                conn.commit();
+            } catch (SQLException e) {
+                throw new CelestaException("Commit unsuccessful: %s", e.getMessage());
```

Comment:
In this case we can't find out a cause of exception. 
Please, let users to see all stacktrace in upper modules.